### PR TITLE
FIX: leaking ioids

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: caproto/_circuit.py [hunk 1]
```diff
@@ -321,6 +321,13 @@ def _process_command(self, role, command):
                                       ClearChannelResponse)):
                 self.channels_sid.pop(chan.sid)
                 self.channels.pop(chan.cid)
+                to_remove = []
+                for k, v in self._ioids.items():
+                    if v is chan:
+                        to_remove.apend(k)
```
Comment (klauer) [inline caproto/_circuit.py]: Typo
```suggestion
                        to_remove.append(k)
```

Or perhaps better:
```suggestion
                    to_remove = [k for k, v in self._ioids.items() if v is chan]
```

File: caproto/tests/test_core.py [hunk 2]
```diff
@@ -367,3 +367,34 @@ def test_circuit_equality():
     b = ca.VirtualCircuit(ca.CLIENT, ('asdf', 1234), 1)
     c = ca.VirtualCircuit(ca.CLIENT, ('asdf', 1234), 2)
     assert a == b != c
+
+
+def test_ioid_exhastion(circuit_pair):
```
Comment (klauer) [inline caproto/tests/test_core.py]:
```suggestion
def test_ioid_exhaustion(circuit_pair):
```

**Comment by tacaswell** [comment]
It takes more than 100s to exhaust the ioids (I think in part because the logging is turned up to 11).

Is there a way to exempt this from the timeouts or the logging?

**Comment by tacaswell** [comment]
ok, temporarily change the constants to be smaller :smiling_imp:

**Pull request merged**